cli/args.go

```diff
@@ -346,41 +350,93 @@ func parseMultiStringArg(args []string, argName string, defaultValue []string) (
 	return stringArgs, nil
 }
 
-// Convert the given variables to a map of environment variables that will expose those variables to Terraform. The
-// keys will be of the format TF_VAR_xxx and the values will be converted to JSON, which Terraform knows how to read
-// natively.
-func toTerraformEnvVars(vars map[string]interface{}) (map[string]string, error) {
-	out := map[string]string{}
+// writeTFVarsFile will create a tfvars file that can be used to invoke the terraform module with the inputs generated
+// in terragrunt.
+func writeTFVarsFile(terragruntOptions *options.TerragruntOptions, terragruntConfig *config.TerragruntConfig) error {
+	terragruntOptions.Logger.Printf("Generating tfvars file")
 
-	for varName, varValue := range vars {
-		envVarName := fmt.Sprintf("TF_VAR_%s", varName)
+	variables, err := terraformModuleVariables(terragruntOptions)
```

Comment:
Sanity check: is it possible that we're somehow concealing a bug by only generating the `.tfvars.json` file with the input vars we find? 

I guess one possibility is that in `terragrunt.hcl`, you have a typo: e.g., you set `instance_typ = "t3.micro"` instead of `instance_type = "t3.micro"`. Terragrunt wouldn't find an `instance_typ` variable in your module's inputs and therefore wouldn't include it in the `.tfvars.json` file. I guess this would be a bit confusing during debugging. That said, today, with env vars, you have the exact same problem, but it's less visible.

If we were to output a `.tfvars` file instead of `tfvars.json`, we could include a commented out section of variables that were in your inputs, but for which there were no matching input vars. That would make things a lot clearer...

Reply:
This is valid, but as you mentioned, we already have this problem, and unlike with env vars, you would spot that the `instance_type` field is missing from the json when you load it. I think the benefit is marginal (e.g., it is just as easy to gloss over comments).

I agree that we should generate `tfvars`, but in the interest of having something ship sooner, I think we should punt on this to the next iteration. Generating `hcl` with comments in a robust manner will be complex since AFAIK, you can't just output the cty; you need to manipulate the AST to inject comment tokens.

Reply:
Roger. Could file issue to track generating a `.tfvars` in future.